test/Tests.Infrastructure/TestBase.cs

```diff
@@ -567,15 +567,18 @@ protected virtual RavenServer GetNewServer(ServerCreationOptions options = null,
                 var hasServerUrls = options.CustomSettings != null && options.CustomSettings.ContainsKey(RavenConfiguration.GetKey(x => x.Core.ServerUrls));
                 var hasDataDirectory = options.CustomSettings != null && options.CustomSettings.ContainsKey(RavenConfiguration.GetKey(x => x.Core.DataDirectory));
                 var hasFeaturesAvailability = options.CustomSettings != null && options.CustomSettings.ContainsKey(RavenConfiguration.GetKey(x => x.Core.FeaturesAvailability));
+                var hasRunInMemory = options.CustomSettings != null && options.CustomSettings.ContainsKey(RavenConfiguration.GetKey(x => x.Core.RunInMemory));
 
                 configuration.Initialize();
 
                 configuration.Logs.Mode = LogMode.None;
                 configuration.Server.Name = ServerName;
-                configuration.Core.RunInMemory = options.RunInMemory;
                 configuration.Server.MaxTimeForTaskToWaitForDatabaseToLoad = new TimeSetting(60, TimeUnit.Seconds);
                 configuration.Licensing.EulaAccepted = true;
 
+                if (hasRunInMemory == false)
+                    configuration.Core.RunInMemory = options.RunInMemory;
```

Comment:
We should forbid setting `RunInMemory` via custom settings then. Throw when this is set and point user that `ServerCreationOptions.RunInMemory` should be used to do that.

It will be better, because now someone can change CustomSettings and then do the opposite in the ServerCreationOptions.

Reply:
Other option (easier?) is to change `ServerCreationOptions.RunInMemory` to nullable and detect that only one of them can be set.

Reply:
What about DataDirectory ? it is also can be set via both ServerCreationOptions & CustomSettings ?

Reply:
https://github.com/ravendb/ravendb/blob/93d8b29d565b2d1245a321d119ee49d9efb2ca39/test/Tests.Infrastructure/TestBase.cs#L582-L598

It is handled, I think?

Reply:
But you mentioned that someone can change CustomSettings and then do the opposite in the ServerCreationOptions, it also apply to DataDirectory, so my question is do we also want to disallow to pass DataDirectory via CustomSettings  ?

Reply:
If you will mark ServerCreationOptions.RunInMemory as nullable. You can allow to set it either via CustomSettings or options, but not both.